compiler/rustc_hir_typeck/src/method/probe.rs

```diff
@@ -1414,7 +1414,21 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
                         },
                         self.param_env,
                         impl_bounds,
-                    ));
+                    )
+                    .collect();
+                    if let Some(obligation) = predicates.iter().find(|obligation| {
```

Comment:
Could this be rewritten as `if !self.infcx.next_trait_solver() && let Some(obligation) = predicates.iter().find(...)`

Reply:
true